core/src/main/java/org/apache/struts2/interceptor/BackgroundProcess.java

```diff
@@ -44,9 +45,8 @@
      * @param invocation The action invocation
      * @param threadPriority The thread priority
      */
-    public BackgroundProcess(String threadName, final ActionInvocation invocation, int threadPriority) {
+    BackgroundProcess(String threadName, final ActionInvocation invocation, int threadPriority) {
```

Comment:
In such case you are reducing re-usability of this class in other interceptors, user will have to copy this class if she wants to re-implement the interceptor

Reply:
Thanks 👍 I did not know user may want. fixed :)

Reply:
Yeah... this is a bit different than development of a normal app. This a framework that should be easily extendable by users and quite often we need to break good practices (like reducing scope) because users would like to use it. You wasn't the first one who did that ;-)